logshifter/input.go

```diff
@@ -64,24 +67,39 @@ func (input *Input) read() {
 
 		copy(cp, line)
 
-		if input.statsChannel != nil {
-			input.statsChannel <- Stat{name: "input.read", value: 1.0}
-		}
+		input.record("input.read", 1.0)
 
 		select {
 		case input.queue <- cp:
 			// queued
+			input.record("input.queue", 1.0)
 		default:
-			// evict the oldest entry to make room
-			<-input.queue
-			if input.statsChannel != nil {
-				input.statsChannel <- Stat{name: "input.drop", value: 1.0}
+			// try to evict the oldest entry to make room
+			select {
+			case <-input.queue:
+				input.record("input.evict", 1.0)
+				// try again to queue
+				select {
+				case input.queue <- cp:
+					// queued
+					input.record("input.queue", 1.0)
+				default:
```

Comment:
I think we both came to the same conclusion here... I couldn't come up with a way it was possible given a single threaded reader, but felt I'd include coverage for the case anyway.